academic-services/knowledge-exploration-service/concepts-queries.md

```diff
@@ -0,0 +1,117 @@
+
+Generally SRGS grammars are context-free, meaning grammar rules can be matched lacking any additional context of previously matched rules. MAKES changes this behavior by including a *grammar context* that enables rules to reference metadata from previously matched rules.
+
+The default SRGS grammar used by MAKES is very simple, with a single top-level rule which is allowed to repeat until all query terms have been processed.
+
+* GetPapers
+  * For current term(s) do *one* of the following:
+    * Match term to paper title words available in the current grammar context
+    * Match terms to paper authors available in the current grammar context
+    * Match terms to paper affiliations available in the current grammar context
+    * Match terms to paper fields of study available in the current grammar context
+    * Match terms to paper conferences available in the current grammar context
+    * Match terms to paper journals available in the current grammar context
+    * Match term to paper publication years available in the current grammar context
+  * If match is found, add the new match to the grammar context. Then either move to the next query segment and repeat top level rule or, if at end of query, move to resolve the hypothesis
+  * If no match is found, terminate the current grammar context
+
+It's important to understand that each hypothesis generates *sub-hypotheses* in the form of an expansion-tree, and each tree branch *gets its own unique copy of the grammar context*. This allows MAKES to generate many potential hypothesis that fully explore the search space for a given query.
+
+## Stage 3: Resolve valid interpretation hypotheses
+
+:::image type="content" source="media/concept-semantic-interpretation-resolve.png" alt-text="MAKES Semantic Interpretation Generation Architecture - Resolve valid hypothesis":::
+
+Once a valid hypothesis has been found, MAKES *resolves* the hypothesis into an interpretation by generating and storing the following:
+
+* A probability score for the hypothesis. This is the static rank of the top-most entity returned by the hypothesis plus any alternative [weight penalties imposed by the SRGS grammar](https://www.w3.org/TR/speech-grammar/#S2.4)
+* XML containing the grammar rules and attributes matched to generate the hypothesis
+* A [structured query expression](concepts-query-expressions.md) that can be used to find all entity results matching the hypothesis
+* Optionally the top matching entities themselves can be also be returned
+
+## Stage 4: Generate interpretation response
+
+The final stage of semantic interpretation generation aggregates all the hypotheses that were found and generates a JSON response.
+
+What is included (i.e. how many results, starting offset, etc.) is controlled by the request parameters, but an example response for the example query "john platt nature 2019" would look like the following:
```

Comment:
Move "what is included...." to after the snapshot. Readers can consume the snapshot first before more information being thrown at them.